Update }bedrock.cube.data.export.pro

wimgielis opened the pull request:
- Sandbox coding was twice there
- More importantly, a view pView should not be recreated when it already exists.


Existing public views should not be recreated. The use case is that views already exist (typically with filters applied) and then a simple call to '}bedrock.cube.data.export' with the name of the view and pFilter = '' would reset the view and remove the filters. That reset is done in '}bedrock.cube.view.create' but probably best to not break into that process. Therefore, a small change in the }bedrock.cube.data.export process.


My main use case why I would suggest this behaviour is the following.

With }bedrock.cube.data.export we can export with filters. However, pFilter does not support MDX subsets. Hence, exporting becomes difficult with }bedrock.cube.data.export as we can only have let's say hardcoded elements or shorter lists of elements.

When we add the possibility to export with filters, the only solution I see is a 3-step approach:
- create a view with Bedrock
- assign subsets to the view, subsets that are the result of MDX queries
- export the view data with Bedrock BUT leave pFilter = '' because we cannot override the filters ánd we supply the view name.

Hence in step 3 Bedrock should not recreate or reset the existing view - otherwise it would throw away the work done in steps 1 and 2.

wimgielis committed:
Update }bedrock.cube.data.export.pro

- Sandbox coding was twice there
- More importantly, a view pView should not be recreated when it already exists. I will explain in more detail in an upcoming pull request.

wimgielis committed:
Exporting MDX views

It is the counterpart of }bedrock.cube.data.export, this process allows for and MDX expression instead of native views.
One limitation to keep the variables order as expected, is that alternate hierarchies in the MDX query are not allowed.

vbhatiacw commented on main/}bedrock.cube.data.export.bymdx.pro:
Correct the Create View by MDX to use cView vs cDefaultView. 
Added note on MDX required to be inserted in 1 row.

vbhatiacw committed:
Update }bedrock.cube.data.export.bymdx.pro

Updated description to text for mdx and updated cDefaultview to cView as part of createview by mdx.
